android/versioned-abis/expoview-abi39_0_0/src/main/java/abi39_0_0/expo/modules/splashscreen/SplashScreenModule.kt

```diff
@@ -9,6 +9,15 @@ import abi39_0_0.org.unimodules.core.errors.CurrentActivityNotFoundException
 import abi39_0_0.org.unimodules.core.interfaces.ActivityProvider
 import abi39_0_0.org.unimodules.core.interfaces.ExpoMethod
 
+// Below import is added explicitly to provide a redirection from versioned code realm to unversioned code realm.
+// Without this import any `SplashScreen.methodName(...)` invocation on JS side ends up in versioned SplashScreen kotlin object that stores no state about the ExperienceActivity.
+// Other solution would be:
+// - create package with the SplashScreeSingletonModule interface and publish as a separate package (that would add complexity into `expo-splash-screen` package itself)
+// - register unversioned SplashScreen object as a singleton module in ModuleRegistry
+// - use ModuleRegistry.getSingletonModule("SplashScreenSingletonModuleName", SplashScreenInterface.class); (that would add complexity into `expo-splash-screen` package itself)
+// TODO (@bbarthec): add this import while performing Android versioning
+import expo.modules.splashscreen.SplashScreen
```

Comment:
For this to be working in future SDK versioning phases you'll need to add this class to [`android-packages-to-keep.txt`](https://github.com/expo/expo/blob/227455e3d6f4bc9d05ca84109db5cbc03a56e342/tools/expotools/src/versioning/android/android-packages-to-keep.txt) and add this import to unversioned `SplashScreenModule` itself.

Reply:
I'm not this happy to trash the module codebase with unnecessary code (import in this case), but rather provide some code injection in versioning process 😉
My argument is: let's keep in mind the module-centric approach and treat Expo Client as a special case that needs more adjustments.

Reply:
If you want the import to be necessary you can always move `SplashScreen` to a separate `.singletons` package. 😛 

You could also add a regexp that replaces
```kt
class SplashScreenModule(context: Context) : ExportedModule(context) {
```
with the same + the import, but you know what they say about problems and regexps. 😃 

Yet another solution is to reference `SplashScreen` everywhere by fully qualified name, i.e. `expo.modules.splashscreen.SplashScreen` — iirc versioning script should catch that (I think this solution is already used somewhere).

--- 

What I mean is — let's solve this problem now. (I also wouldn't say that adding an import is _trashing the module with unnecessary code_, but that's an opinion.)

Reply:
> What I mean is — let's solve this problem now. 

I agree with this and would also add that we should go with whatever solution is **most likely to keep working in the future**. The cleanliness of the codebase is subjective and is less important than the direct maintenance cost to us to keep the module working in future SDK versions.